File: tests/unit/java/com/trulioo/sdk/model/ResultTest.java
package com.trulioo.sdk.model;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.*;

import java.util.List;
import java.util.Arrays;

public class ResultTest {
    private Result result;

    @BeforeEach
    public void beforeEach() {
        result = new Result();
    }

    @Test
    public void testIndex() {
        String index = "1";
        result.setIndex(index);

        assertEquals(index, result.getIndex());
        assertEquals(index, new Result().index(index).getIndex());
    }

    @Test
    public void testBusinessName() {
        String businessName = "test";
        result.setBusinessName(businessName);

        assertEquals(businessName, result.getBusinessName());
        assertEquals(businessName, new Result().businessName(businessName).getBusinessName());
    }

    @Test
    public void testMatchingScore() {
        String matchingScore = "100%";
        result.setMatchingScore(matchingScore);

        assertEquals(matchingScore, result.getMatchingScore());
        assertEquals(matchingScore, new Result().matchingScore(matchingScore).getMatchingScore());
    }

    @Test
    public void testBusinessRegistrationNumber() {
        String businessRegistrationNumber = "123";
        result.setBusinessRegistrationNumber(businessRegistrationNumber);

        assertEquals(businessRegistrationNumber, result.getBusinessRegistrationNumber());
        assertEquals(businessRegistrationNumber,
                new Result().businessRegistrationNumber(businessRegistrationNumber).getBusinessRegistrationNumber());
    }

    @Test
    public void testDuNSNumber() {
        String duNSNumber = "123";
        result.setDuNSNumber(duNSNumber);

        assertEquals(duNSNumber, result.getDuNSNumber());
        assertEquals(duNSNumber, new Result().duNSNumber(duNSNumber).getDuNSNumber());
    }

    @Test
    public void testBusinessTaxIDNumber() {
        String businessTaxIDNumber = "123";
        result.setBusinessTaxIDNumber(businessTaxIDNumber);

        assertEquals(businessTaxIDNumber, result.getBusinessTaxIDNumber());
        assertEquals(businessTaxIDNumber,
                new Result().businessTaxIDNumber(businessTaxIDNumber).getBusinessTaxIDNumber());
    }

    @Test
    public void testBusinessLicenseNumber() {
        String businessLicenseNumber = "123";
        result.setBusinessLicenseNumber(businessLicenseNumber);

        assertEquals(businessLicenseNumber, result.getBusinessLicenseNumber());
        assertEquals(businessLicenseNumber,
                new Result().businessLicenseNumber(businessLicenseNumber).getBusinessLicenseNumber());
    }

    @Test
    public void testJurisdictionOfIncorporation() {
        String jurisdictionOfIncorporation = "Alberta";
        result.setJurisdictionOfIncorporation(jurisdictionOfIncorporation);

        assertEquals(jurisdictionOfIncorporation, result.getJurisdictionOfIncorporation());
        assertEquals(jurisdictionOfIncorporation,
                new Result().jurisdictionOfIncorporation(jurisdictionOfIncorporation).getJurisdictionOfIncorporation());
    }

    @Test
    public void testFullAddress() {
        String fullAddress = "123 Seasame St";
        result.setFullAddress(fullAddress);

        assertEquals(fullAddress, result.getFullAddress());
        assertEquals(fullAddress, new Result().fullAddress(fullAddress).getFullAddress());
    }

    @Test
    public void testBusinessStatus() {
        String businessStatus = "status";
        result.setBusinessStatus(businessStatus);

        assertEquals(businessStatus, result.getBusinessStatus());
        assertEquals(businessStatus, new Result().businessStatus(businessStatus).getBusinessStatus());
    }

    @Test
    public void testTradeStyleName() {
        String tradeStyleName = "name";
        result.setTradeStyleName(tradeStyleName);

        assertEquals(tradeStyleName, result.getTradeStyleName());
        assertEquals(tradeStyleName, new Result().tradeStyleName(tradeStyleName).getTradeStyleName());
    }

    @Test
    public void testBusinessType() {
        String businessType = "type";
        result.setBusinessType(businessType);

        assertEquals(businessType, result.getBusinessType());
        assertEquals(businessType, new Result().businessType(businessType).getBusinessType());
    }

    @Test
    public void testAddress() {
        Address address = new Address().buildingNumber("123");
        result.setAddress(address);

        assertEquals(address, result.getAddress());
        assertEquals(address, new Result().address(address).getAddress());
    }

    @Test
    public void testOtherBusinessNames() {
        result.addOtherBusinessNamesItem("abc");
        result.addOtherBusinessNamesItem("123");
        List<String> otherBusinessNames = Arrays.asList("other");
        result.setOtherBusinessNames(otherBusinessNames);

        assertEquals(otherBusinessNames, result.getOtherBusinessNames());
        assertEquals(otherBusinessNames, new Result().otherBusinessNames(otherBusinessNames).getOtherBusinessNames());
    }

    @Test
    public void testWebsite() {
        String website = "https://trulioo.com";
        result.setWebsite(website);

        assertEquals(website, result.getWebsite());
        assertEquals(website, new Result().website(website).getWebsite());
    }

    @Test
    public void testTelephone() {
        String telephone = "18887730179";
        result.setTelephone(telephone);

        assertEquals(telephone, result.getTelephone());
        assertEquals(telephone, new Result().telephone(telephone).getTelephone());
    }

    @Test
    public void testTaxIDNumber() {
        String taxIDNumber = "123";
        result.setTaxIDNumber(taxIDNumber);

        assertEquals(taxIDNumber, result.getTaxIDNumber());
        assertEquals(taxIDNumber, new Result().taxIDNumber(taxIDNumber).getTaxIDNumber());
    }

    @Test
    public void testTaxIDNumbers() {
        result.addTaxIDNumbersItem("123");
        result.addTaxIDNumbersItem("456");
        List<String> taxIDNumbers = Arrays.asList("789");
        result.setTaxIDNumbers(taxIDNumbers);

        assertEquals(taxIDNumbers, result.getTaxIDNumbers());
        assertEquals(taxIDNumbers, new Result().taxIDNumbers(taxIDNumbers).getTaxIDNumbers());
    }

    @Test
    public void testEmailAddress() {
        String emailAddress = "support@trulioo.com";
        result.setEmailAddress(emailAddress);

        assertEquals(emailAddress, result.getEmailAddress());
        assertEquals(emailAddress, new Result().emailAddress(emailAddress).getEmailAddress());
    }

    @Test
    public void testWebDomain() {
        String webDomain = "trulioo.com";
        result.setWebDomain(webDomain);

        assertEquals(webDomain, result.getWebDomain());
        assertEquals(webDomain, new Result().webDomain(webDomain).getWebDomain());
    }

    @Test
    public void testWebDomains() {
        result.addWebDomainsItem("google.com");
        result.addWebDomainsItem("amazon.com");
        List<String> webDomains = Arrays.asList("trulioo.com");
        result.setWebDomains(webDomains);

        assertEquals(webDomains, result.getWebDomains());
        assertEquals(webDomains, new Result().webDomains(webDomains).getWebDomains());
    }

    @Test
    public void testNAICS() {
        result.addNAICSItem(new BusinessSearchResponseIndustryCode());
        result.addNAICSItem(new BusinessSearchResponseIndustryCode());
        List<BusinessSearchResponseIndustryCode> NAICS = Arrays
                .asList(new BusinessSearchResponseIndustryCode().code("123"));
        result.setNAICS(NAICS);

        assertEquals(NAICS, result.getNAICS());
        assertEquals(NAICS, new Result().NAICS(NAICS).getNAICS());
    }

    @Test
    public void testSIC() {
        result.addSICItem(new BusinessSearchResponseIndustryCode());
        result.addSICItem(new BusinessSearchResponseIndustryCode());
        List<BusinessSearchResponseIndustryCode> SIC = Arrays
                .asList(new BusinessSearchResponseIndustryCode().code("123"));
        result.setSIC(SIC);

        assertEquals(SIC, result.getSIC());
        assertEquals(SIC, new Result().SIC(SIC).getSIC());
    }

    @Test
    public void testEquals() {
        String index = "1";
        Result result1 = new Result().index(index);

        assertEquals(result1, result1);
        assertEquals(result1, new Result().index(index));
        assertNotEquals(result1, new Result().index(index + "1"));
        assertNotEquals(result1, null);
    }

    @Test
    public void testHashcode() {
        String index = "1";

        assertEquals(new Result().index(index).hashCode(), new Result().index(index).hashCode());
    }

    @Test
    public void testToString() {
        String index = "1";

        assertEquals(new Result().index(index).toString(), new Result().index(index).toString());
    }
}

File: tests/unit/php/Model/ResultTest.php
<?php

namespace Trulioo\SDK\Test\Model;

use PHPUnit\Framework\TestCase;
use Trulioo\SDK\Model\Result;
use Trulioo\SDK\Model\Address;
use Trulioo\SDK\Model\BusinessSearchResponseIndustryCode;

class ResultTest extends TestCase
{
    private Result $result;

    public function setUp(): void
    {
        $this->result = new Result([]);
    }

    public function testPropertyIndex()
    {
        $this->result->setIndex('abc');
        $this->assertEquals('abc', $this->result->getIndex());
    }

    public function testPropertyBusinessName()
    {
        $this->result->setBusinessName('test');
        $this->assertEquals('test', $this->result->getBusinessName());
    }

    public function testPropertyMatchingScore()
    {
        $this->result->setMatchingScore('100');
        $this->assertEquals('100', $this->result->getMatchingScore());
    }

    public function testPropertyBusinessRegistrationNumber()
    {
        $this->result->setBusinessRegistrationNumber('123');
        $this->assertEquals('123', $this->result->getBusinessRegistrationNumber());
    }

    public function testPropertyDunsNumber()
    {
        $this->result->setDunsNumber('123');
        $this->assertEquals('123', $this->result->getDunsNumber());
    }

    public function testPropertyBusinessTaxIdNumber()
    {
        $this->result->setBusinessTaxIdNumber('123');
        $this->assertEquals('123', $this->result->getBusinessTaxIdNumber());
    }

    public function testPropertyBusinessLicenseNumber()
    {
        $this->result->setBusinessLicenseNumber('123');
        $this->assertEquals('123', $this->result->getBusinessLicenseNumber());
    }

    public function testPropertyJurisdictionOfIncorporation()
    {
        $this->result->setJurisdictionOfIncorporation('Alberta');
        $this->assertEquals('Alberta', $this->result->getJurisdictionOfIncorporation());
    }

    public function testPropertyFullAddress()
    {
        $this->result->setFullAddress('123 Seasame St');
        $this->assertEquals('123 Seasame St', $this->result->getFullAddress());
    }

    public function testPropertyBusinessStatus()
    {
        $this->result->setBusinessStatus('status');
        $this->assertEquals('status', $this->result->getBusinessStatus());
    }

    public function testPropertyTradeStyleName()
    {
        $this->result->setTradeStyleName('name');
        $this->assertEquals('name', $this->result->getTradeStyleName());
    }

    public function testPropertyBusinessType()
    {
        $this->result->setBusinessType('Company');
        $this->assertEquals('Company', $this->result->getBusinessType());
    }

    public function testPropertyAddress()
    {
        $this->result->setAddress(new Address([
            'building_number' => '10'
        ]));
        $this->assertEquals(
            new Address([
                'building_number' => '10'
            ]),
            $this->result->getAddress()
        );
    }

    public function testPropertyOtherBusinessNames()
    {
        $this->result->setOtherBusinessNames(['test2']);
        $this->assertEquals(['test2'], $this->result->getOtherBusinessNames());
    }

    public function testPropertyWebsite()
    {
        $this->result->setWebsite('https://trulioo.com');
        $this->assertEquals('https://trulioo.com', $this->result->getWebsite());
    }

    public function testPropertyTelephone()
    {
        $this->result->setTelephone('18887730179');
        $this->assertEquals('18887730179', $this->result->getTelephone());
    }

    public function testPropertyTaxIdNumber()
    {
        $this->result->setTaxIdNumber('123');
        $this->assertEquals('123', $this->result->getTaxIdNumber());
    }

    public function testPropertyTaxIdNumbers()
    {
        $this->result->setTaxIdNumbers(['123']);
        $this->assertEquals(['123'], $this->result->getTaxIdNumbers());
    }

    public function testPropertyEmailAddress()
    {
        $this->result->setEmailAddress('support@trulioo.com');
        $this->assertEquals('support@trulioo.com', $this->result->getEmailAddress());
    }

    public function testPropertyWebDomain()
    {
        $this->result->setWebDomain('trulioo.com');
        $this->assertEquals('trulioo.com', $this->result->getWebDomain());
    }

    public function testPropertyWebDomains()
    {
        $this->result->setWebDomains(['trulioo.com']);
        $this->assertEquals(['trulioo.com'], $this->result->getWebDomains());
    }

    public function testPropertyNaics()
    {
        $this->result->setNaics([
            new BusinessSearchResponseIndustryCode([
                'code' => 'abc'
            ])
        ]);
        $this->assertEquals(
            [
                new BusinessSearchResponseIndustryCode([
                    'code' => 'abc'
                ])
            ],
            $this->result->getNaics()
        );
    }

    public function testPropertySic()
    {
        $this->result->setSic([
            new BusinessSearchResponseIndustryCode([
                'code' => 'abc'
            ])
        ]);
        $this->assertEquals(
            [
                new BusinessSearchResponseIndustryCode([
                    'code' => 'abc'
                ])
            ],
            $this->result->getSic()
        );
    }
}
